### PR TITLE
Plans: Add highlighting for designated product features

### PR DESCRIPTION
**Pull request by mattgawarecki**
Resolves `1200412004370260-as-1200788956056656`.

#### Changes proposed in this Pull Request

* Add support for an `isHighlighted` function to force highlighting for a given feature on all product cards.
* Add highlighting for some features on the storage upgrade page.
* Add support for an `isDifferentiator` function to cause some feature items to be bulleted with a star instead of a standard bullet point.
* Add differentiator styling to some features on the 1 TB upgrade product on the storage upgrade page.
* Make the description for Backup (1 TB) the same as Backup (10 GB).

##### Non-storage pricing pages

1. Visit the Plans/Pricing page in your chosen UI:
  a. Calypso Green: `/pricing?flags=jetpack/only-realtime-products` or `/pricing/:site?flags=jetpack/only-realtime-products`
  b. Calypso Blue: `/plans/:site?flags=jetpack/only-realtime-products`
2. Verify that no features are bolded or starred on this page, and that all copy is the same as in production.

##### Site has not purchased a 1 TB upgrade

1. Navigate to the storage upgrade page in your chosen UI:
  a. Calypso Green: `/pricing/storage/:site?flags=jetpack/only-realtime-products`
  b. Calypso Blue: `/plans/storage/:site?flags=jetpack/only-realtime-products`
2. Verify that both the 10GB and 1TB cards are visible, and that the top few features are bolded on both.
3. Verify that on the 1TB card, the top few features are denoted with stars as their bullet points instead of filled circles.

##### Site has purchased a 1 TB upgrade

1. Follow the same steps as in the previous case.
2. Note that only the 1 TB product is visible on the page, and that it consumes the full section width.
3. Verify the same features are bolded and starred as in the previous case.

#### Reference captures

Design mock: `1a5ZL6xHHfOXdpXYQdiZuT-fi-774%3A0`

##### Standard pricing pages (no highlighting)

image image

##### Site has Backup (10 GB)

https://user-images.githubusercontent.com/670067/136833444-d9be957e-ffd3-46a9-8661-8afd7b4d3b5e.mp4

##### Site has Backup (1 TB)

https://user-images.githubusercontent.com/670067/136833416-6ce762f9-fc11-41fc-be97-bcadb5c07fd9.mp4

### CODE REVIEW

File: client/lib/plans/features-list.js [hunk 1]
```diff
@@ -1074,7 +1089,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_PRODUCT_SEARCH_V2 ]: {
 		getSlug: () => FEATURE_PRODUCT_SEARCH_V2,
-		getTitle: () => i18n.translate( 'Site Search: up to 100k records' ),
+		getTitle: () => i18n.translate( 'Search up to 100k records' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/lib/plans/features-list.js [hunk 2]
```diff
@@ -1103,7 +1118,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_CRM_V2 ]: {
 		getSlug: () => FEATURE_CRM_V2,
-		getTitle: () => i18n.translate( 'CRM: Entrepreneur bundle' ),
+		getTitle: () => i18n.translate( 'CRM Entrepreneur bundle' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **8** times:
`translate( 'CRM: Entrepreneur' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/highlighted-jetpack-product-features)

File: client/lib/plans/features-list.js [hunk 3]
```diff
@@ -1315,6 +1331,69 @@ export const FEATURES_LIST = {
...
+		getSlug: () => FEATURE_TIERED_ONE_CLICK_RESTORES_1_YEAR,
+		getTitle: () => i18n.translate( 'One-click restore from the last year' ),
+	},
+	[ FEATURE_TIERED_ONE_CLICK_RESTORES_UNLIMITED ]: {
+		getSlug: () => FEATURE_TIERED_ONE_CLICK_RESTORES_UNLIMITED,
+		getTitle: () => i18n.translate( 'One-click restores from any time' ),
+	},
+	[ FEATURE_TIERED_PRODUCT_BACKUP ]: {
+		getSlug: () => FEATURE_TIERED_PRODUCT_BACKUP,
+		getTitle: () => i18n.translate( 'All Backup features' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js:1378]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'All Jetpack Backup features' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/highlighted-jetpack-product-features)

File: client/lib/plans/features-list.js [hunk 4]
```diff
@@ -1315,6 +1331,69 @@ export const FEATURES_LIST = {
...
+		isHighlighted: () => isEnabled( 'jetpack/only-realtime-products' ),
+	},
+	[ FEATURE_TIERED_PRODUCT_SCAN ]: {
+		getSlug: () => FEATURE_TIERED_PRODUCT_SCAN,
+		getTitle: () => i18n.translate( 'Scan (real-time & automated)' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Scan (real-time, automated)' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/highlighted-jetpack-product-features)

File: client/lib/plans/features-list.js [hunk 5]
```diff
@@ -1315,6 +1331,69 @@ export const FEATURES_LIST = {
...
+		isHighlighted: () => isEnabled( 'jetpack/only-realtime-products' ),
+	},
+	[ FEATURE_TIERED_PRODUCT_SECURITY ]: {
+		getSlug: () => FEATURE_TIERED_PRODUCT_SECURITY,
+		getTitle: () => i18n.translate( 'All Security features' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'All Security Daily features' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/highlighted-jetpack-product-features)

File: client/lib/plans/features-list.js [hunk 6]
```diff
@@ -1314,6 +1330,69 @@ export const FEATURES_LIST = {
...
+		getTitle: () => i18n.translate( 'Backups are cloud based' ),
+		isHighlighted: () => isEnabled( 'jetpack/only-realtime-products' ),
+	},
+	[ FEATURE_JETPACK_20GB_BACKUP_STORAGE ]: {
+		getSlug: () => FEATURE_JETPACK_20GB_BACKUP_STORAGE,
+		getTitle: () => i18n.translate( '20GB of backup storage' ),
+	},
+	[ FEATURE_JETPACK_1TB_BACKUP_STORAGE ]: {
+		getSlug: () => FEATURE_JETPACK_1TB_BACKUP_STORAGE,
+		getTitle: () => i18n.translate( '1TB of backup storage' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( '1TB of storage' )` *ES Score: 7*

File: client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+						{
+							text: translate( '%(storageAmount)s backup storage', { args: { storageAmount } } ),
+							isHighlighted: true,
+						},
+						{
+							text: translate( 'One-click restore from the last 30 days of backups' ),
+							isHighlighted: true,
+						},
+						{
+							text: translate( '30-day activity log' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Activity log: 30-day archive' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/highlighted-jetpack-product-features)

File: client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+							isHighlighted: true,
+						},
+						{
+							text: translate( 'Real-time backups (as you edit)' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Real-time backups &mdash; we save as you edit' )` *ES Score: 6*

**Review by eeeeevon13 — CHANGES_REQUESTED** [review]
I'd go with a green star so it's not directly associated with the sale language (in yellow). Also please ensure we are using the star icon from the WordPress design library (not material design).



--



--

**Comment by mattgawarecki** [comment]
Updated. Here's a screenshot of what it should look like now, with the star's color set to Jetpack Green 50%.

image

**Review by monsieur-z — APPROVED** [review]
LGTM!

**Pull request merged**